src/qibolab/platforms/abstract.py

```diff
@@ -175,58 +175,88 @@ def transpile(self, circuit: Circuit):
         for qubit in range(circuit.nqubits):
             sequence.virtual_z_phases[qubit] = 0
```

Comment:
For example you can completely remove this if you use `collections.defaultdict(int)`.

Reply:
Thanks Stavros,
I saw you using it in your latest code for QM. I hesitated whether to use it or not.
When importing libraries there are **clear advantages** but also some disadvantages:
- some other developers may not be familiar with them
- it introduces objects that are similar to those built in, but that may not be accepted by some other functions, which could lead to errors when casting between object types
- it requires importing additional libraries that increase the loading time
- if those libraries are not part of python standard library, then it increases the number of dependencies of the project

I guess we have some kind of unwritten guidelines about when to make use of libraries.

In this particular case, `collections` is part of python standard library, and I suppose you would argue that we all should be familiar with it, so I will make use of it from now on.

Reply:
Thanks for explaining. The main motivations here were:
* we save a few lines of code and the less lines we maintain the better,
* it looks like a perfect use case for `defaultdict` and if this object was not useful then they wouldn't have it in standard Python.

> * some other developers may not be familiar with them

True, but at least standard libraries are well documented so if they Google the object they will find it within minutes and they will know it in their next job.

> * it introduces objects that are similar to those built in, but that may not be accepted by some other functions, which could lead to errors when casting between object types

I am not sure but I hope this is not the case for `defaultdict`. If we notice any issues in the future we can change.

> * it requires importing additional libraries that increase the loading time

Indeed, for example importing TensorFlow takes a couple of seconds which is not negligible. But for `collections` and we should not notice any overhead other than standard Python.

> * if those libraries are not part of python standard library, then it increases the number of dependencies of the project

For non-standard libraries, I also hesitate using them for this reason. Adding dependencies means that we should also care if they are well maintained.